### DISCUSSION TITLE
Fix for Docker v26.0.1 and above

### DISCUSSION THREAD

**Discussion by Jip-Hop** [Jip-Hop/jailmaker#148] [Announcements]
Users who are still using host networking will run into issues when they upgrade docker to v26.0.1 and above. I recommend anyone using docker inside their jail to stop using the `docker_compatible` config option a.s.a.p. and start using macvlan or bridge networking. Refer to the readme for [the new recommended setup to run docker in a jail](https://github.com/Jip-Hop/jailmaker?tab=readme-ov-file#docker). The new setup is more secure and because the jail has its own networking namespace it's less likely to run into compatibility problems ([which is what's causing the docker issues](https://github.com/moby/moby/issues/47751)). For a smooth migration I recommend to follow these [migration instructions](https://github.com/Jip-Hop/jailmaker/discussions/137#discussioncomment-9220929). My docker jail is using bridge networking and the latest docker v26.1.0 is running perfectly!

**Comment by neoKushan** [comment]
Looking into this issue, it seems v26.1.1 potentially fixes the underlying change? 

https://github.com/moby/moby/releases/tag/v26.1.1

> Resolve an issue preventing container creation on hosts with a read-only /proc/sys/net filesystem. If IPv6 cannot be disabled on an interface due to this, either disable IPv6 by default on the host or ensure /proc/sys/net is read-write. Otherwise, start dockerd with DOCKER_ALLOW_IPV6_ON_IPV4_INTERFACE=1 to bypass the error.

Am I correct that this is the underlying cause, or am I way off?

**Reply by Jip-Hop** [reply]
I think so, but I still recommend bridge or macvlan networking over host networking to prevent these issues in the future.
